zscaler/icon_zscaler/actions/blacklist_url/action.py

```diff
@@ -23,6 +23,9 @@ def run(self, params={}):
         urls = params.get(Input.URLS)
         normalized_urls = []
         for url in urls:
+            if url and not url.startswith("http"):
+                url = f"http://{url}"
```

Comment:
We should blacklist both the https and http version of the URL (2 URLs)

Reply:
Lookup and blacklist works only for domain. Not url. So we add here http to work with urlparse which get domain part from url. Scheme is not passing to API.

```
lookup_urls.append(
    urlparse(url).hostname
)
```

Reply:
Shouldn't this be something like to support a user entering both https and https URL and turning them into domain?

```
if url and not (url.startswith("http") or url.startswith("https")):
  url = f"http://{url}"
```

Reply:
@jschipp-r7 This code support http and https, user can provide both and also user can provide domain.

Reply:
oh, duh `http` is in `https`. Ha, thanks - I had a moment there.